controller/api/destination/server.go

```diff
@@ -364,17 +364,44 @@ func (s *server) GetProfile(dest *pb.GetDestination, stream pb.Destination_GetPr
 	return nil
 }
 
-func (s *server) createAddress(pod *corev1.Pod, port uint32) (watcher.Address, error) {
+// getPortForPod returns the port that a `pod` is listening on.
+//
+// If the `host` address given is the HostIP, use the HostPort mapping for the `pod`
+// to determine the port to return.
+func (s *server) getPortForPod(pod *corev1.Pod, host string, port uint32) (uint32, error) {
+	if host == pod.Status.PodIP {
```

Comment:
We probably should be using the `PodIPs` list...

https://github.com/kubernetes/api/blob/9e1815a99d4fa3983f39ae3bf1cd68dda5df49da/core/v1/types.go#L4037

Though I'm sure we're not doing that everywhere else we need to be doing that...